### PR TITLE
[move][test scenario] Fixed docs. Fixed immutable modified bug

### PR DESCRIPTION
**Pull request by tnowacki**
- Fixed test scenario references in docs
- Updated immutable object usage in test scenario to check for modifications

### CODE REVIEW

File: crates/sui-framework/src/natives/test_scenario.rs [hunk 1]
```diff
@@ -209,8 +210,22 @@ pub fn end_transaction(
         })
         .collect::<BTreeMap<_, _>>();
     // update inventories
-    for (id, (_, value)) in new_object_values {
+    // check for bad updates to immutable values
+    for (id, (ty, value)) in new_object_values {
```
Comment (amnn) [inline crates/sui-framework/src/natives/test_scenario.rs:214]: nit: merge the line comments into one?

```
// update inventories, checking for bad updates to immutable values
```

File: doc/src/build/programming-with-objects/ch1-object-basics.md [hunk 2]
```diff
@@ -105,7 +105,8 @@ First of all, we begin the test with a hardcoded test address, which will also g
 ```rust
 let owner = @0x1;
 // Create a ColorObject and transfer it to @owner.
-let scenario = &mut test_scenario::begin(&owner);
```
Comment (amnn) [inline doc/src/build/programming-with-objects/ch1-object-basics.md:108]: :facepalm: I didn't spot these docs needed changing, thanks for catching.  Should we also include these changes in the update to the docs site for today? cc @randall-Mysten

**Review by amnn — APPROVED** [review]

File: doc/src/build/programming-with-objects/ch1-object-basics.md (hunk 2 above)
Comment (randall-Mysten) [reply doc/src/build/programming-with-objects/ch1-object-basics.md:108]: I can push another update once this PR is merged

File: doc/src/build/move/build-test.md [hunk 3]
```diff
@@ -266,34 +266,45 @@ function.
...
+            init(test_scenario::ctx(scenario));
+        };
+        // second transaction executed by admin to create the sword
+        test_scenario::next_tx(scenario, admin);
+        {
+            let forge = test_scenario::take_from_sender<Forge>(scenario);
             // create the sword and transfer it to the initial owner
-            sword_create(42, 7, initial_owner, test_scenario::ctx(scenario));
+            sword_create(&mut forge, 42, 7, initial_owner, test_scenario::ctx(scenario));
+            test_scenario::return_to_sender(scenario, forge)
```
Comment (3rror404) [inline doc/src/build/move/build-test.md:286]: `sword_create()` doesn't take a `Forge` as a parameter. I think all that is needed here is:

```
sword_create(42, 7, initial_owner, test_scenario::ctx(scenario));
```

**Pull request merged**